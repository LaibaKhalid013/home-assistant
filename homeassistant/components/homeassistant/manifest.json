{
  "domain": "homeassistant",
  "name": "Home Assistant Core Integration",
  "after_dependencies": ["repairs"],
  "codeowners": ["@home-assistant/core"],
  "documentation": "https://www.home-assistant.io/integrations/homeassistant",
  "integration_type": "system",
  "quality_scale": "internal"
}
