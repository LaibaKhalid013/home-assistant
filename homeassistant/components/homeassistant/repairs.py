"""Repairs for Home Assistant."""

from __future__ import annotations

from homeassistant import data_entry_flow
from homeassistant.components.repairs import ConfirmRepairFlow, RepairsFlow
from homeassistant.core import HomeAssistant


class IntegrationNotFoundFlow(RepairsFlow):
    """Handler for an issue fixing flow."""

    def __init__(self, data: dict[str, str]) -> None:
        """Initialize."""
        self.description_placeholders: dict[str, str] = {
            "device_name": data["device_name"]
        }

    async def async_step_init(
        self, user_input: dict[str, str] | None = None
    ) -> data_entry_flow.FlowResult:
        """Handle the first step of a fix flow."""
        return self.async_show_menu(
            menu_options=["confirm", "ignore"],
            description_placeholders=self.description_placeholders,
        )


async def async_create_fix_flow(
    hass: HomeAssistant, issue_id: str, data: dict[str, str] | None
) -> RepairsFlow:
    """Create flow."""

    if issue_id.split(".")[0] == "integration_not_found":
        assert data
        return IntegrationNotFoundFlow(data)
    return ConfirmRepairFlow()
