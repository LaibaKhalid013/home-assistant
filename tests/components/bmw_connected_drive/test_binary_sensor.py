"""Test BMW binary sensors."""

from freezegun import freeze_time
import respx
from syrupy.assertion import SnapshotAssertion

from homeassistant.core import HomeAssistant

# from bimmer_connected.vehicle.reports import ConditionBasedServiceReport
from . import setup_mocked_integration


@freeze_time("2023-06-22 10:30:00+00:00")
async def test_entity_state_attrs(
    hass: HomeAssistant,
    bmw_fixture: respx.Router,
    snapshot: SnapshotAssertion,
) -> None:
    """Test binary sensor states and attributes."""

    # Setup component
    assert await setup_mocked_integration(hass)

    # Get all select entities
    assert hass.states.async_all("binary_sensor") == snapshot
