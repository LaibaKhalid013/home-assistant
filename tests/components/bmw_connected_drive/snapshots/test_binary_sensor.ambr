# serializer version: 1
# name: test_entity_state_attrs
  list([
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'iX xDrive50',
        'device_class': 'opening',
        'friendly_name': 'iX xDrive50 Lids',
        'hood': 'CLOSED',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'sunRoof': 'CLOSED',
        'trunk': 'CLOSED',
        'vin': 'WBA00000000DEMO01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_lids',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'iX xDrive50',
        'device_class': 'opening',
        'friendly_name': 'iX xDrive50 Windows',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'vin': 'WBA00000000DEMO01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_windows',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'iX xDrive50',
        'device_class': 'lock',
        'door_lock_state': 'LOCKED',
        'friendly_name': 'iX xDrive50 Door lock state',
        'vin': 'WBA00000000DEMO01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_door_lock_state',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'brake_fluid': 'OK',
        'brake_fluid_date': '2024-12-01',
        'brake_fluid_distance': '50000 km',
        'car': 'iX xDrive50',
        'device_class': 'problem',
        'friendly_name': 'iX xDrive50 Condition based services',
        'tire_wear_front': 'OK',
        'tire_wear_rear': 'OK',
        'vehicle_check': 'OK',
        'vehicle_check_date': '2024-12-01',
        'vehicle_check_distance': '50000 km',
        'vehicle_tuv': 'OK',
        'vehicle_tuv_date': '2024-12-01',
        'vehicle_tuv_distance': '50000 km',
        'vin': 'WBA00000000DEMO01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_condition_based_services',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'iX xDrive50',
        'device_class': 'problem',
        'friendly_name': 'iX xDrive50 Check control messages',
        'tire_pressure': 'LOW',
        'vin': 'WBA00000000DEMO01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_check_control_messages',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'device_class': 'battery_charging',
        'friendly_name': 'iX xDrive50 Charging status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_charging_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'device_class': 'plug',
        'friendly_name': 'iX xDrive50 Connection status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_connection_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'friendly_name': 'iX xDrive50 Pre entry climatization',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.ix_xdrive50_pre_entry_climatization',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i4 eDrive40',
        'device_class': 'opening',
        'friendly_name': 'i4 eDrive40 Lids',
        'hood': 'CLOSED',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'trunk': 'CLOSED',
        'vin': 'WBA00000000DEMO02',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_lids',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i4 eDrive40',
        'device_class': 'opening',
        'friendly_name': 'i4 eDrive40 Windows',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'vin': 'WBA00000000DEMO02',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_windows',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i4 eDrive40',
        'device_class': 'lock',
        'door_lock_state': 'LOCKED',
        'friendly_name': 'i4 eDrive40 Door lock state',
        'vin': 'WBA00000000DEMO02',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_door_lock_state',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'brake_fluid': 'OK',
        'brake_fluid_date': '2024-12-01',
        'brake_fluid_distance': '50000 km',
        'car': 'i4 eDrive40',
        'device_class': 'problem',
        'friendly_name': 'i4 eDrive40 Condition based services',
        'tire_wear_front': 'OK',
        'tire_wear_rear': 'OK',
        'vehicle_check': 'OK',
        'vehicle_check_date': '2024-12-01',
        'vehicle_check_distance': '50000 km',
        'vehicle_tuv': 'OK',
        'vehicle_tuv_date': '2024-12-01',
        'vehicle_tuv_distance': '50000 km',
        'vin': 'WBA00000000DEMO02',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_condition_based_services',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i4 eDrive40',
        'device_class': 'problem',
        'friendly_name': 'i4 eDrive40 Check control messages',
        'tire_pressure': 'LOW',
        'vin': 'WBA00000000DEMO02',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_check_control_messages',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'device_class': 'battery_charging',
        'friendly_name': 'i4 eDrive40 Charging status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_charging_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'device_class': 'plug',
        'friendly_name': 'i4 eDrive40 Connection status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_connection_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'friendly_name': 'i4 eDrive40 Pre entry climatization',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i4_edrive40_pre_entry_climatization',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'M340i xDrive',
        'device_class': 'opening',
        'friendly_name': 'M340i xDrive Lids',
        'hood': 'CLOSED',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'trunk': 'CLOSED',
        'vin': 'WBA00000000DEMO03',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.m340i_xdrive_lids',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'M340i xDrive',
        'device_class': 'opening',
        'friendly_name': 'M340i xDrive Windows',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'vin': 'WBA00000000DEMO03',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.m340i_xdrive_windows',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'M340i xDrive',
        'device_class': 'lock',
        'door_lock_state': 'LOCKED',
        'friendly_name': 'M340i xDrive Door lock state',
        'vin': 'WBA00000000DEMO03',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.m340i_xdrive_door_lock_state',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'brake_fluid': 'OK',
        'brake_fluid_date': '2024-12-01',
        'brake_fluid_distance': '50000 km',
        'car': 'M340i xDrive',
        'device_class': 'problem',
        'friendly_name': 'M340i xDrive Condition based services',
        'oil': 'OK',
        'oil_date': '2024-12-01',
        'oil_distance': '50000 km',
        'tire_wear_front': 'OK',
        'tire_wear_rear': 'OK',
        'vehicle_check': 'OK',
        'vehicle_check_date': '2024-12-01',
        'vehicle_check_distance': '50000 km',
        'vehicle_tuv': 'OK',
        'vehicle_tuv_date': '2024-12-01',
        'vehicle_tuv_distance': '50000 km',
        'vin': 'WBA00000000DEMO03',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.m340i_xdrive_condition_based_services',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'M340i xDrive',
        'device_class': 'problem',
        'engine_oil': 'LOW',
        'friendly_name': 'M340i xDrive Check control messages',
        'tire_pressure': 'LOW',
        'vin': 'WBA00000000DEMO03',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.m340i_xdrive_check_control_messages',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i3 (+ REX)',
        'device_class': 'opening',
        'friendly_name': 'i3 (+ REX) Lids',
        'hood': 'CLOSED',
        'leftFront': 'CLOSED',
        'leftRear': 'CLOSED',
        'rightFront': 'CLOSED',
        'rightRear': 'CLOSED',
        'sunRoof': 'CLOSED',
        'trunk': 'CLOSED',
        'vin': 'WBY00000000REXI01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_lids',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i3 (+ REX)',
        'device_class': 'opening',
        'friendly_name': 'i3 (+ REX) Windows',
        'leftFront': 'CLOSED',
        'rightFront': 'CLOSED',
        'vin': 'WBY00000000REXI01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_windows',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i3 (+ REX)',
        'device_class': 'lock',
        'door_lock_state': 'UNLOCKED',
        'friendly_name': 'i3 (+ REX) Door lock state',
        'vin': 'WBY00000000REXI01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_door_lock_state',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'brake_fluid': 'OK',
        'brake_fluid_date': '2022-10-01',
        'car': 'i3 (+ REX)',
        'device_class': 'problem',
        'friendly_name': 'i3 (+ REX) Condition based services',
        'vehicle_check': 'OK',
        'vehicle_check_date': '2023-05-01',
        'vehicle_tuv': 'OK',
        'vehicle_tuv_date': '2023-05-01',
        'vin': 'WBY00000000REXI01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_condition_based_services',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'car': 'i3 (+ REX)',
        'device_class': 'problem',
        'friendly_name': 'i3 (+ REX) Check control messages',
        'vin': 'WBY00000000REXI01',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_check_control_messages',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'device_class': 'battery_charging',
        'friendly_name': 'i3 (+ REX) Charging status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_charging_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'device_class': 'plug',
        'friendly_name': 'i3 (+ REX) Connection status',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_connection_status',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'on',
    }),
    StateSnapshot({
      'attributes': ReadOnlyDict({
        'attribution': 'Data provided by MyBMW',
        'friendly_name': 'i3 (+ REX) Pre entry climatization',
      }),
      'context': <ANY>,
      'entity_id': 'binary_sensor.i3_rex_pre_entry_climatization',
      'last_changed': <ANY>,
      'last_reported': <ANY>,
      'last_updated': <ANY>,
      'state': 'off',
    }),
  ])
# ---
